{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 182,
   "id": "e6793c49",
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "import urllib\n",
    "from urllib import request, parse, error\n",
    "import requests\n",
    "import base64\n",
    "import numpy\n",
    "from scipy.stats import chisquare\n",
    "from scipy.stats import fisher_exact"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 213,
   "id": "9a1cc98f",
   "metadata": {},
   "outputs": [],
   "source": [
    "def Query_Panther(genes):\n",
    "    url = \"https://www.pantherdb.org/services/oai/pantherdb/enrich/overrep?geneInputList=\" \\\n",
    "        + genes \\\n",
    "        + \"&organism=9606&annotDataSet=GO%3A0003674&enrichmentTestType=FISHER&correction=NONE\"\n",
    "    request = urllib.request.Request(url)\n",
    "    response = urllib.request.urlopen(request)\n",
    "    data = json.loads(response.read())\n",
    "    return data\n",
    "#0003674 molecular function\n",
    "#0008150 biological process\n",
    "#0005575 cellular component"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 214,
   "id": "75d151b7",
   "metadata": {},
   "outputs": [],
   "source": [
    "def Retrieve_Panther_Results(inputGenes):\n",
    "    goDict = {}\n",
    "    genes = \"\"\n",
    "    geneCounter = 0\n",
    "    mappedGenes = 0\n",
    "    for line in inputGenes:\n",
    "        genes = genes + \",\" + line.rstrip()\n",
    "        geneCounter += 1\n",
    "    \n",
    "        if geneCounter > 4000:\n",
    "            data = Query_Panther(genes)\n",
    "            for entry in data[\"results\"][\"result\"]:\n",
    "                if entry[\"term\"][\"label\"] not in goDict:\n",
    "                    goDict[entry[\"term\"][\"label\"]] = 0\n",
    "                goDict[entry[\"term\"][\"label\"]] += entry[\"number_in_list\"]\n",
    "            mappedGenes += data[\"results\"][\"input_list\"][\"mapped_count\"]\n",
    "            genes = \"\"\n",
    "            geneCounter = 0\n",
    "            \n",
    "    data = Query_Panther(genes)\n",
    "    mappedGenes += data[\"results\"][\"input_list\"][\"mapped_count\"]\n",
    "    for entry in data[\"results\"][\"result\"]:\n",
    "        if entry[\"term\"][\"label\"] not in goDict:\n",
    "            goDict[entry[\"term\"][\"label\"]] = 0\n",
    "        goDict[entry[\"term\"][\"label\"]] += entry[\"number_in_list\"]   \n",
    "    return goDict, mappedGenes\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 259,
   "id": "75e7d377",
   "metadata": {},
   "outputs": [],
   "source": [
    "cell = \"MCF-7\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 260,
   "id": "3fc44842",
   "metadata": {},
   "outputs": [],
   "source": [
    "dataDir = \"/home/moorej3/Lab/ENCODE/Encyclopedia/V7/Registry/V7-hg38/3D-Links/PLS-comparisons/\"\n",
    "input1 = open(dataDir+\"Panther-Common-\" + cell + \".txt\")\n",
    "goDict1, count1 = Retrieve_Panther_Results(input1)\n",
    "input1.close()\n",
    "\n",
    "input2 = open(dataDir+\"Panther-RNAPII-\" + cell + \".txt\")\n",
    "goDict2, count2 = Retrieve_Panther_Results(input2)\n",
    "input2.close()\n",
    "\n",
    "input3 = open(dataDir+\"Panther-CTCF-\" + cell + \".txt\")\n",
    "goDict3, count3 = Retrieve_Panther_Results(input3)\n",
    "input3.close()\n",
    "\n",
    "input4 = open(dataDir+\"Panther-None-\" + cell + \".txt\")\n",
    "goDict4, count4 = Retrieve_Panther_Results(input4)\n",
    "input4.close()\n",
    "\n",
    "input5 = open(dataDir+\"Panther-Background-\" + cell + \".txt\")\n",
    "goDict5, count5 = Retrieve_Panther_Results(input5)\n",
    "input5.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 254,
   "id": "d02fca2a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "transmembrane_signaling_receptor_activity \t 5.38e-08 \t 1.13 \t 0.6 \t 2.94 \t 1.72\n"
     ]
    }
   ],
   "source": [
    "term = \"transmembrane signaling receptor activity\"\n",
    "fraction = goDict5[term]/count5\n",
    "observed = [goDict1[term], goDict2[term], goDict3[term], goDict4[term]]\n",
    "expected = [count1 * fraction, count2 * fraction, count3 * fraction, count4 * fraction] \n",
    "p = chisquare(observed, expected)[1]\n",
    "ratio = numpy.divide(numpy.array(observed),numpy.array(expected))\n",
    "termNew = term.replace(\" \",\"_\")\n",
    "padj = p * len(goDict1)\n",
    "print(termNew, \"\\t\", '{:.2e}'.format(padj), \"\\t\", round(ratio[0],2), \"\\t\", round(ratio[1],2), \"\\t\", round(ratio[2],2), \"\\t\", round(ratio[3],2))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 261,
   "id": "91bff58d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "nucleic_acid_binding \t 9.98e-15 \t 0.95 \t 1.19 \t 0.59 \t 0.88\n"
     ]
    }
   ],
   "source": [
    "term = \"nucleic acid binding\"\n",
    "fraction = goDict5[term]/count5\n",
    "observed = [goDict1[term], goDict2[term], goDict3[term], goDict4[term]]\n",
    "expected = [count1 * fraction, count2 * fraction, count3 * fraction, count4 * fraction] \n",
    "p = chisquare(observed, expected)[1]\n",
    "ratio = numpy.divide(numpy.array(observed),numpy.array(expected))\n",
    "termNew = term.replace(\" \",\"_\")\n",
    "padj = p * len(goDict1)\n",
    "print(termNew, \"\\t\", '{:.2e}'.format(padj), \"\\t\", round(ratio[0],2), \"\\t\", round(ratio[1],2), \"\\t\", round(ratio[2],2), \"\\t\", round(ratio[3],2))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 262,
   "id": "c684e1d2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "organic_cyclic_compound_binding \t 1.07e-09 \t 0.98 \t 1.11 \t 0.74 \t 0.93\n",
      "heterocyclic_compound_binding \t 1.09e-10 \t 0.99 \t 1.11 \t 0.71 \t 0.92\n",
      "G_protein-coupled_receptor_activity \t 6.79e-06 \t 0.72 \t 0.7 \t 1.85 \t 1.52\n",
      "RNA_binding \t 5.79e-19 \t 1.01 \t 1.26 \t 0.48 \t 0.72\n",
      "nucleic_acid_binding \t 1.97e-18 \t 0.95 \t 1.19 \t 0.59 \t 0.88\n",
      "transmembrane_signaling_receptor_activity \t 2.76e-12 \t 0.75 \t 0.68 \t 1.85 \t 1.47\n",
      "molecular_transducer_activity \t 8.94e-13 \t 0.82 \t 0.72 \t 1.9 \t 1.37\n",
      "signaling_receptor_activity \t 8.94e-13 \t 0.82 \t 0.72 \t 1.9 \t 1.37\n",
      "catalytic_activity,_acting_on_a_nucleic_acid \t 5.17e-06 \t 1.14 \t 1.09 \t 0.18 \t 0.84\n",
      "cadherin_binding \t 1.80e-06 \t 1.13 \t 1.35 \t 0.62 \t 0.61\n",
      "signaling_receptor_regulator_activity \t 1.03e-11 \t 1.34 \t 0.54 \t 2.78 \t 1.09\n",
      "signaling_receptor_activator_activity \t 1.38e-11 \t 1.33 \t 0.53 \t 2.85 \t 1.04\n",
      "receptor_ligand_activity \t 3.30e-12 \t 1.35 \t 0.51 \t 2.94 \t 1.01\n",
      "DNA_binding \t 2.34e-07 \t 0.89 \t 1.15 \t 0.67 \t 0.99\n",
      "structural_constituent_of_ribosome \t 7.31e-07 \t 0.83 \t 1.54 \t 0.63 \t 0.43\n",
      "voltage-gated_monoatomic_cation_channel_activity \t 3.49e-19 \t 0.51 \t 0.22 \t 4.83 \t 1.72\n",
      "monoatomic_ion_gated_channel_activity \t 9.14e-19 \t 0.46 \t 0.55 \t 3.58 \t 1.52\n",
      "gated_channel_activity \t 4.29e-18 \t 0.48 \t 0.56 \t 3.53 \t 1.5\n",
      "channel_activity \t 6.65e-16 \t 0.59 \t 0.71 \t 2.87 \t 1.37\n",
      "passive_transmembrane_transporter_activity \t 5.39e-16 \t 0.58 \t 0.71 \t 2.86 \t 1.38\n",
      "monoatomic_ion_channel_activity \t 4.73e-13 \t 0.59 \t 0.7 \t 2.74 \t 1.39\n",
      "ligand-gated_monoatomic_ion_channel_activity \t 4.12e-09 \t 0.22 \t 0.63 \t 4.03 \t 1.49\n",
      "monoatomic_cation_channel_activity \t 6.02e-13 \t 0.59 \t 0.66 \t 3.11 \t 1.35\n",
      "signaling_receptor_binding"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-262-d5acc506d369>:7: RuntimeWarning: invalid value encountered in true_divide\n",
      "  ratio = numpy.divide(numpy.array(observed),numpy.array(expected))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " \t 1.24e-06 \t 1.15 \t 0.79 \t 1.37 \t 1.16\n",
      "ligand-gated_channel_activity \t 8.40e-09 \t 0.22 \t 0.67 \t 3.95 \t 1.46\n",
      "voltage-gated_potassium_channel_activity \t 1.34e-21 \t 0.5 \t 0.32 \t 6.71 \t 1.29\n",
      "voltage-gated_monoatomic_ion_channel_activity \t 1.25e-18 \t 0.46 \t 0.44 \t 4.34 \t 1.56\n",
      "voltage-gated_channel_activity \t 3.33e-18 \t 0.5 \t 0.43 \t 4.29 \t 1.55\n",
      "potassium_channel_regulator_activity \t 7.23e-06 \t 0.83 \t 1.17 \t 4.97 \t 0.72\n",
      "potassium_channel_activity \t 2.28e-19 \t 0.47 \t 0.4 \t 5.8 \t 1.35\n",
      "ligand-gated_monoatomic_cation_channel_activity \t 1.55e-08 \t 0.17 \t 0.68 \t 4.29 \t 1.35\n",
      "inorganic_cation_transmembrane_transporter_activity \t 4.86e-11 \t 0.88 \t 0.69 \t 2.46 \t 1.17\n",
      "growth_factor_activity \t 6.94e-06 \t 1.68 \t 0.52 \t 2.46 \t 0.86\n",
      "salt_transmembrane_transporter_activity \t 3.48e-09 \t 0.86 \t 0.69 \t 2.25 \t 1.24\n",
      "transmembrane_transporter_activity \t 1.05e-07 \t 0.96 \t 0.79 \t 1.8 \t 1.11\n",
      "monoatomic_ion_transmembrane_transporter_activity \t 2.26e-09 \t 0.86 \t 0.75 \t 2.15 \t 1.18\n",
      "metal_ion_transmembrane_transporter_activity \t 3.06e-11 \t 0.8 \t 0.61 \t 2.51 \t 1.33\n",
      "monoatomic_cation_transmembrane_transporter_activity \t 1.29e-09 \t 0.87 \t 0.71 \t 2.3 \t 1.17\n",
      "inorganic_molecular_entity_transmembrane_transporter_activity \t 1.33e-09 \t 0.89 \t 0.73 \t 2.24 \t 1.15\n",
      "transporter_activity \t 5.12e-08 \t 0.95 \t 0.8 \t 1.74 \t 1.12\n",
      "potassium_ion_transmembrane_transporter_activity \t 1.22e-14 \t 0.63 \t 0.47 \t 4.36 \t 1.43\n"
     ]
    }
   ],
   "source": [
    "#outputFile=open(\"test-MCF.txt\", \"w+\")\n",
    "for term in goDict5:\n",
    "    fraction = goDict5[term]/count5\n",
    "    observed = [goDict1[term], goDict2[term], goDict3[term], goDict4[term]]\n",
    "    expected = [count1 * fraction, count2 * fraction, count3 * fraction, count4 * fraction] \n",
    "    p = chisquare(observed, expected)[1]\n",
    "    ratio = numpy.divide(numpy.array(observed),numpy.array(expected))\n",
    "    termNew = term.replace(\" \",\"_\")\n",
    "    if p < correctedP and goDict5[term] >= 25:\n",
    "        print(termNew, \"\\t\", '{:.2e}'.format(p), \"\\t\", round(ratio[0],2), \"\\t\", round(ratio[1],2), \"\\t\", round(ratio[2],2), \"\\t\", round(ratio[3],2))\n",
    "#outputFile.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 211,
   "id": "c112d4c4",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-211-3572270caaee>:5: RuntimeWarning: invalid value encountered in true_divide\n",
      "  ratio = numpy.divide(numpy.array(observed),numpy.array(expected))\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "regulation_of_nucleobase-containing_compound_metabolic_process \t 3.02e-07 \t 0.53 \t 0.23 \t 0.28\n",
      "regulation_of_gene_expression \t 8.84e-07 \t 0.54 \t 0.26 \t 0.31\n",
      "regulation_of_RNA_metabolic_process \t 5.02e-07 \t 0.52 \t 0.21 \t 0.26\n",
      "regulation_of_DNA-templated_transcription \t 1.38e-06 \t 0.52 \t 0.19 \t 0.23\n",
      "regulation_of_RNA_biosynthetic_process \t 2.22e-06 \t 0.53 \t 0.19 \t 0.23\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-211-3572270caaee>:5: RuntimeWarning: divide by zero encountered in true_divide\n",
      "  ratio = numpy.divide(numpy.array(observed),numpy.array(expected))\n"
     ]
    }
   ],
   "source": [
    "for term in goDict5:\n",
    "    observed = [goDict1[term], count1-goDict1[term]]\n",
    "    expected = [goDict2[term], count2-goDict2[term]]\n",
    "    p = fisher_exact([observed,expected])[1]\n",
    "    ratio = numpy.divide(numpy.array(observed),numpy.array(expected))\n",
    "    termNew = term.replace(\" \",\"_\")\n",
    "    if p < correctedP and goDict5[term] >= 25:\n",
    "        print(termNew, \"\\t\", '{:.2e}'.format(p), \"\\t\", round(ratio[0],2), \"\\t\", round(goDict1[term]/count1,2), \"\\t\", round(goDict2[term]/count2,2))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 267,
   "id": "caca413b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "organic_cyclic_compound_binding \t 1.87e-07 \t 27.93 \t 37.84\n",
      "heterocyclic_compound_binding \t 8.09e-09 \t 26.58 \t 37.47\n",
      "RNA_binding \t 3.51e-07 \t 5.56 \t 11.54\n",
      "nucleic_acid_binding \t 1.25e-10 \t 15.02 \t 25.62\n",
      "catalytic_activity,_acting_on_a_nucleic_acid \t 2.45e-07 \t 0.75 \t 4.25\n",
      "signaling_receptor_regulator_activity \t 3.57e-06 \t 4.35 \t 1.57\n",
      "catalytic_activity,_acting_on_RNA \t 9.89e-06 \t 0.3 \t 2.68\n",
      "signaling_receptor_activator_activity \t 3.51e-06 \t 4.2 \t 1.48\n",
      "receptor_ligand_activity \t 2.08e-06 \t 4.2 \t 1.43\n",
      "voltage-gated_monoatomic_cation_channel_activity \t 1.77e-07 \t 2.85 \t 0.59\n",
      "monoatomic_ion_gated_channel_activity \t 3.62e-08 \t 4.35 \t 1.22\n",
      "gated_channel_activity \t 4.76e-08 \t 4.35 \t 1.23\n",
      "channel_activity \t 5.27e-08 \t 5.71 \t 1.99\n",
      "passive_transmembrane_transporter_activity \t 5.85e-08 \t 5.71 \t 2.0\n",
      "monoatomic_ion_channel_activity \t 1.49e-06 \t 4.8 \t 1.75\n",
      "monoatomic_cation_channel_activity \t 7.60e-07 \t 4.2 \t 1.35\n",
      "voltage-gated_potassium_channel_activity \t 6.01e-09 \t 2.7 \t 0.4\n",
      "voltage-gated_monoatomic_ion_channel_activity \t 4.92e-08 \t 3.45 \t 0.8\n",
      "voltage-gated_channel_activity \t 5.85e-08 \t 3.45 \t 0.81\n",
      "potassium_channel_activity \t 1.59e-08 \t 2.85 \t 0.49\n",
      "inorganic_cation_transmembrane_transporter_activity \t 5.47e-07 \t 6.31 \t 2.56\n",
      "monoatomic_ion_transmembrane_transporter_activity \t 4.71e-06 \t 6.91 \t 3.21\n",
      "metal_ion_transmembrane_transporter_activity \t 8.29e-06 \t 4.8 \t 1.91\n",
      "monoatomic_cation_transmembrane_transporter_activity \t 3.49e-06 \t 6.16 \t 2.68\n",
      "inorganic_molecular_entity_transmembrane_transporter_activity \t 2.28e-06 \t 6.76 \t 3.02\n",
      "potassium_ion_transmembrane_transporter_activity \t 3.43e-07 \t 3.0 \t 0.69\n"
     ]
    }
   ],
   "source": [
    "for term in goDict5:\n",
    "    observed = [goDict3[term], count3-goDict3[term]]\n",
    "    expected = [goDict5[term], count5-goDict5[term]]\n",
    "    p = fisher_exact([observed,expected])[1]\n",
    "    termNew = term.replace(\" \",\"_\")\n",
    "    if p < correctedP and goDict5[term] >= 25:\n",
    "        print(termNew, \"\\t\", '{:.2e}'.format(p), \"\\t\", round(goDict3[term]/count3*100,2), \"\\t\", round(goDict5[term]/count5*100,2))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 269,
   "id": "8b296650",
   "metadata": {},
   "outputs": [],
   "source": [
    "dataDir = \"/home/moorej3/Lab/ENCODE/Encyclopedia/V7/Registry/V7-hg38/Cell-Type-Specific/\"\n",
    "input1 = open(dataDir+\"low-low.genes\")\n",
    "goDict1, count1 = Retrieve_Panther_Results(input1)\n",
    "input1.close()\n",
    "\n",
    "input2 = open(dataDir+\"low-high.genes\")\n",
    "goDict2, count2 = Retrieve_Panther_Results(input2)\n",
    "input2.close()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 272,
   "id": "5ba50146",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "RNA_binding \t 2.82e+03 \t 4.4 \t 2.69\n",
      "cytoskeletal_protein_binding \t 2.84e+03 \t 16.69 \t 10.2\n",
      "UNCLASSIFIED \t 1.20e+03 \t 10.02 \t 6.12\n",
      "molecular_function \t 1.20e+03 \t 153.57 \t 93.88\n",
      "binding \t 2.13e+03 \t 142.94 \t 87.38\n",
      "protein_binding \t 2.82e+03 \t 126.86 \t 77.55\n",
      "actin_binding \t 4.12e+03 \t 7.74 \t 4.73\n",
      "catalytic_activity,_acting_on_a_nucleic_acid \t 5.06e+03 \t 1.37 \t 0.83\n",
      "ion_binding \t 1.65e+03 \t 59.18 \t 36.18\n",
      "kinesin_binding \t 4.81e+02 \t 1.82 \t 1.11\n",
      "tubulin_binding \t 3.53e+03 \t 6.37 \t 3.9\n",
      "gated_channel_activity \t 3.97e+03 \t 5.77 \t 3.53\n",
      "RNA_polymerase_II_cis-regulatory_region_sequence-specific_DNA_binding \t 8.15e+01 \t 14.72 \t 9.0\n",
      "monoatomic_ion_gated_channel_activity \t 4.51e+03 \t 5.61 \t 3.43\n",
      "cis-regulatory_region_sequence-specific_DNA_binding \t 6.48e+01 \t 14.87 \t 9.09\n",
      "monoatomic_cation_channel_activity \t 1.99e+03 \t 5.61 \t 3.43\n",
      "channel_activity \t 1.36e+03 \t 7.59 \t 4.64\n",
      "passive_transmembrane_transporter_activity \t 1.36e+03 \t 7.59 \t 4.64\n",
      "DNA-binding_transcription_factor_activity \t 1.58e+02 \t 16.69 \t 10.2\n",
      "potassium_ion_transmembrane_transporter_activity \t 2.37e+03 \t 3.34 \t 2.04\n",
      "metal_ion_binding \t 1.69e+03 \t 42.64 \t 26.07\n",
      "DNA-binding_transcription_factor_activity,_RNA_polymerase_II-specific \t 2.80e+02 \t 16.08 \t 9.83\n",
      "monoatomic_ion_channel_activity \t 1.84e+03 \t 6.68 \t 4.08\n",
      "RNA_polymerase_II_transcription_regulatory_region_sequence-specific_DNA_binding \t 1.45e+02 \t 16.24 \t 9.93\n",
      "protein_kinase_activity \t 2.43e+03 \t 8.04 \t 4.92\n",
      "cation_binding \t 2.34e+03 \t 42.94 \t 26.25\n",
      "calcium_ion_binding \t 3.43e+03 \t 9.56 \t 5.84\n",
      "protein_domain_specific_binding \t 3.75e+03 \t 8.95 \t 5.47\n",
      "transmembrane_transporter_binding \t 2.17e+03 \t 2.88 \t 1.76\n",
      "protein_kinase_binding \t 1.93e+03 \t 9.41 \t 5.75\n",
      "phosphatidylinositol-4,5-bisphosphate_binding \t 6.53e+02 \t 2.12 \t 1.3\n",
      "kinase_binding \t 3.04e+03 \t 10.17 \t 6.22\n",
      "transcription_cis-regulatory_region_binding \t 1.00e+02 \t 16.84 \t 10.3\n",
      "structural_constituent_of_ribosome \t 1.92e+03 \t 0.0 \t 0.0\n",
      "transcription_regulatory_region_nucleic_acid_binding \t 1.00e+02 \t 16.84 \t 10.3\n",
      "phosphatidylinositol_bisphosphate_binding \t 2.83e+02 \t 2.43 \t 1.48\n",
      "channel_regulator_activity \t 5.06e+03 \t 3.03 \t 1.86\n",
      "sequence-specific_double-stranded_DNA_binding \t 8.55e+01 \t 17.3 \t 10.58\n",
      "G_protein-coupled_receptor_activity \t 1.75e+03 \t 3.64 \t 2.23\n",
      "microtubule_binding \t 5.06e+03 \t 4.4 \t 2.69\n",
      "phospholipid_binding \t 9.21e+02 \t 6.68 \t 4.08\n",
      "kinase_activity \t 2.61e+03 \t 9.56 \t 5.84\n",
      "anion_binding \t 3.43e+03 \t 25.19 \t 15.4\n",
      "voltage-gated_channel_activity \t 5.06e+03 \t 3.49 \t 2.13\n",
      "DNA-binding_transcription_activator_activity,_RNA_polymerase_II-specific \t 6.91e+02 \t 6.53 \t 3.99\n",
      "amyloid-beta_binding \t 3.24e+03 \t 1.97 \t 1.21\n",
      "catalytic_activity,_acting_on_RNA \t 5.06e+03 \t 1.06 \t 0.65\n",
      "nucleic_acid_binding \t 6.17e+02 \t 25.34 \t 15.49\n",
      "mRNA_binding \t 3.73e+03 \t 0.76 \t 0.46\n",
      "phosphotransferase_activity,_alcohol_group_as_acceptor \t 3.30e+03 \t 8.65 \t 5.29\n",
      "DNA-binding_transcription_activator_activity \t 6.91e+02 \t 6.53 \t 3.99\n",
      "salt_transmembrane_transporter_activity \t 3.20e+03 \t 7.59 \t 4.64\n",
      "ion_channel_regulator_activity \t 4.30e+03 \t 2.73 \t 1.67\n",
      "double-stranded_DNA_binding \t 9.41e+01 \t 17.75 \t 10.85\n",
      "protein_tyrosine_kinase_activity \t 2.06e+03 \t 2.73 \t 1.67\n",
      "metal_ion_transmembrane_transporter_activity \t 5.06e+03 \t 6.07 \t 3.71\n",
      "ephrin_receptor_binding \t 3.80e+03 \t 1.06 \t 0.65\n",
      "sequence-specific_DNA_binding \t 1.84e+02 \t 17.75 \t 10.85\n",
      "purine_ribonucleoside_triphosphate_binding \t 2.71e+03 \t 19.27 \t 11.78\n",
      "potassium_channel_activity \t 2.64e+03 \t 2.43 \t 1.48\n",
      "monoatomic_cation_transmembrane_transporter_activity \t 2.36e+03 \t 7.59 \t 4.64\n",
      "monoatomic_ion_transmembrane_transporter_activity \t 2.53e+03 \t 8.8 \t 5.38\n",
      "protein_kinase_C_binding \t 1.13e+03 \t 1.37 \t 0.83\n",
      "GTPase_regulator_activity \t 5.08e+02 \t 6.53 \t 3.99\n",
      "voltage-gated_monoatomic_ion_channel_activity \t 5.06e+03 \t 3.34 \t 2.04\n",
      "nucleoside-triphosphatase_regulator_activity \t 5.08e+02 \t 6.53 \t 3.99\n",
      "potassium_channel_regulator_activity \t 1.13e+03 \t 1.37 \t 0.83\n",
      "transferase_activity,_transferring_phosphorus-containing_groups \t 1.29e+03 \t 10.77 \t 6.59\n",
      "channel_inhibitor_activity \t 5.06e+03 \t 1.21 \t 0.74\n",
      "purine_ribonucleotide_binding \t 3.26e+03 \t 19.73 \t 12.06\n",
      "transcription_regulator_activity \t 8.12e+01 \t 19.73 \t 12.06\n",
      "molecular_function_regulator_activity \t 1.30e+03 \t 22.0 \t 13.45\n",
      "PDZ_domain_binding \t 5.06e+03 \t 1.82 \t 1.11\n",
      "actin_filament_binding \t 5.06e+03 \t 3.34 \t 2.04\n",
      "extracellular_matrix_binding \t 1.13e+03 \t 1.37 \t 0.83\n",
      "ribonucleotide_binding \t 3.26e+03 \t 19.73 \t 12.06\n",
      "purine_nucleotide_binding \t 3.86e+03 \t 20.49 \t 12.52\n",
      "inorganic_molecular_entity_transmembrane_transporter_activity \t 2.46e+03 \t 8.19 \t 5.01\n",
      "inorganic_cation_transmembrane_transporter_activity \t 3.58e+03 \t 6.98 \t 4.27\n",
      "collagen_binding \t 4.07e+03 \t 1.52 \t 0.93\n",
      "lipid_binding \t 1.96e+03 \t 9.71 \t 5.94\n",
      "ligand-gated_monoatomic_cation_channel_activity \t 4.16e+03 \t 2.12 \t 1.3\n",
      "ligand-gated_monoatomic_ion_channel_activity \t 5.06e+03 \t 2.58 \t 1.58\n",
      "ligand-gated_channel_activity \t 5.06e+03 \t 2.58 \t 1.58\n",
      "microtubule_motor_activity \t 4.02e+03 \t 1.52 \t 0.93\n",
      "catalytic_activity,_acting_on_DNA \t 3.32e+03 \t 0.61 \t 0.37\n",
      "voltage-gated_monoatomic_cation_channel_activity \t 5.06e+03 \t 2.58 \t 1.58\n",
      "protein_serine_kinase_activity \t 1.80e+03 \t 4.86 \t 2.97\n",
      "cytoskeletal_motor_activity \t 4.16e+03 \t 2.12 \t 1.3\n",
      "transcription_coregulator_binding \t 1.13e+03 \t 2.12 \t 1.3\n",
      "clathrin_binding \t 1.98e+03 \t 1.52 \t 0.93\n",
      "phosphatidylinositol_phosphate_binding \t 1.24e+02 \t 2.88 \t 1.76\n",
      "ion_channel_inhibitor_activity \t 5.06e+03 \t 1.06 \t 0.65\n",
      "enzyme_regulator_activity \t 1.98e+03 \t 14.11 \t 8.63\n",
      "ATP_binding \t 1.72e+03 \t 15.48 \t 9.46\n",
      "SH2_domain_binding \t 3.80e+03 \t 1.06 \t 0.65\n",
      "protein_serine/threonine_kinase_activity \t 1.99e+03 \t 5.46 \t 3.34\n",
      "carbohydrate_derivative_binding \t 3.63e+03 \t 22.46 \t 13.73\n",
      "nuclease_activity \t 5.06e+03 \t 0.46 \t 0.28\n",
      "enzyme_binding \t 2.58e+03 \t 20.79 \t 12.71\n",
      "growth_factor_binding \t 2.19e+03 \t 2.28 \t 1.39\n",
      "voltage-gated_potassium_channel_activity \t 3.12e+03 \t 1.82 \t 1.11\n",
      "ribonucleoprotein_complex_binding \t 5.06e+03 \t 0.3 \t 0.19\n",
      "nucleotide_binding \t 4.17e+03 \t 21.24 \t 12.99\n",
      "adenyl_ribonucleotide_binding \t 1.98e+03 \t 15.78 \t 9.65\n",
      "transcription_factor_binding \t 1.19e+02 \t 7.13 \t 4.36\n",
      "nucleoside_phosphate_binding \t 4.17e+03 \t 21.24 \t 12.99\n",
      "catalytic_activity,_acting_on_a_tRNA \t 5.06e+03 \t 0.15 \t 0.09\n",
      "phosphatidylinositol_binding \t 1.48e+03 \t 3.79 \t 2.32\n",
      "transcription_coactivator_binding \t 2.51e+03 \t 1.06 \t 0.65\n",
      "calcium_channel_activity \t 5.06e+03 \t 2.12 \t 1.3\n",
      "guanyl-nucleotide_exchange_factor_activity \t 7.47e+01 \t 3.19 \t 1.95\n",
      "adenyl_nucleotide_binding \t 2.56e+03 \t 16.54 \t 10.11\n",
      "protein-containing_complex_binding \t 2.10e+03 \t 17.6 \t 10.76\n",
      "phosphatidylserine_binding \t 5.06e+03 \t 1.21 \t 0.74\n",
      "transmembrane_receptor_protein_tyrosine_kinase_activity \t 3.97e+03 \t 1.21 \t 0.74\n",
      "ATP-dependent_activity,_acting_on_DNA \t 5.06e+03 \t 0.15 \t 0.09\n",
      "transmembrane_transporter_activity \t 3.52e+03 \t 11.68 \t 7.14\n",
      "enzyme_activator_activity \t 5.06e+03 \t 7.13 \t 4.36\n",
      "nuclear_estrogen_receptor_binding \t 1.33e+03 \t 0.91 \t 0.56\n",
      "calcium-dependent_phospholipid_binding \t 2.51e+03 \t 1.06 \t 0.65\n",
      "scaffold_protein_binding \t 3.01e+03 \t 1.21 \t 0.74\n",
      "catalytic_activity,_acting_on_a_protein \t 3.90e+03 \t 22.0 \t 13.45\n",
      "phosphoric_diester_hydrolase_activity \t 5.06e+03 \t 1.52 \t 0.93\n",
      "beta-catenin_binding \t 3.18e+03 \t 1.52 \t 0.93\n",
      "phosphoric_ester_hydrolase_activity \t 2.82e+03 \t 4.4 \t 2.69\n",
      "coreceptor_activity \t 3.80e+03 \t 1.06 \t 0.65\n",
      "signaling_receptor_binding \t 1.08e+03 \t 14.87 \t 9.09\n",
      "DNA-binding_transcription_repressor_activity \t 5.06e+03 \t 4.1 \t 2.5\n",
      "non-membrane_spanning_protein_tyrosine_kinase_activity \t 1.33e+03 \t 0.91 \t 0.56\n",
      "calcium_ion_transmembrane_transporter_activity \t 5.06e+03 \t 2.12 \t 1.3\n",
      "GTPase_activator_activity \t 3.68e+03 \t 3.49 \t 2.13\n",
      "signaling_adaptor_activity \t 1.98e+03 \t 1.52 \t 0.93\n",
      "molecular_adaptor_activity \t 3.01e+03 \t 6.22 \t 3.8\n",
      "phosphoprotein_phosphatase_activity \t 3.52e+03 \t 2.43 \t 1.48\n",
      "ribosome_binding \t 5.06e+03 \t 0.15 \t 0.09\n",
      "transmembrane_receptor_protein_kinase_activity \t 3.11e+03 \t 1.37 \t 0.83\n",
      "phosphotyrosine_residue_binding \t 5.06e+03 \t 0.91 \t 0.56\n",
      "DNA-binding_transcription_repressor_activity,_RNA_polymerase_II-specific \t 4.42e+03 \t 3.95 \t 2.41\n",
      "GTPase_activity \t 3.81e+03 \t 3.95 \t 2.41\n",
      "oxidoreductase_activity \t 1.83e+03 \t 3.95 \t 2.41\n",
      "SNARE_binding \t 5.06e+03 \t 1.67 \t 1.02\n",
      "phospholipase_activity \t 5.06e+03 \t 1.67 \t 1.02\n",
      "damaged_DNA_binding \t 5.06e+03 \t 0.0 \t 0.0\n",
      "cargo_receptor_activity \t 2.79e+03 \t 1.37 \t 0.83\n",
      "structural_molecule_activity \t 6.24e+02 \t 4.4 \t 2.69\n",
      "growth_factor_receptor_binding \t 5.06e+03 \t 1.97 \t 1.21\n",
      "protein_self-association \t 1.70e+03 \t 1.21 \t 0.74\n",
      "protein_phosphatase_binding \t 2.37e+03 \t 1.97 \t 1.21\n",
      "myosin_binding \t 3.01e+03 \t 1.21 \t 0.74\n",
      "small_molecule_binding \t 3.68e+03 \t 23.67 \t 14.47\n",
      "transporter_activity \t 4.68e+03 \t 12.14 \t 7.42\n",
      "identical_protein_binding \t 4.46e+03 \t 20.18 \t 12.34\n",
      "helicase_activity \t 5.06e+03 \t 0.46 \t 0.28\n",
      "RNA_nuclease_activity \t 5.06e+03 \t 0.3 \t 0.19\n",
      "endonuclease_activity \t 5.06e+03 \t 0.3 \t 0.19\n",
      "phosphatidylinositol-3,4,5-trisphosphate_binding \t 8.27e+02 \t 0.76 \t 0.46\n",
      "protein_serine/threonine_phosphatase_activity \t 4.07e+03 \t 1.52 \t 0.93\n",
      "structural_constituent_of_chromatin \t 5.06e+03 \t 0.0 \t 0.0\n",
      "exonuclease_activity,_active_with_either_ribo-_or_deoxyribonucleic_acids_and_producing_5'-phosphomonoesters \t 5.06e+03 \t 0.0 \t 0.0\n",
      "2-oxoglutarate-dependent_dioxygenase_activity \t 7.27e+02 \t 0.0 \t 0.0\n",
      "protein_dimerization_activity \t 3.44e+03 \t 10.47 \t 6.4\n",
      "RNA_methyltransferase_activity \t 5.06e+03 \t 0.0 \t 0.0\n",
      "rRNA_binding \t 5.06e+03 \t 0.0 \t 0.0\n",
      "protein_folding_chaperone \t 5.06e+03 \t 0.0 \t 0.0\n",
      "DNA-binding_transcription_factor_binding \t 3.20e+02 \t 5.31 \t 3.25\n",
      "tau_protein_binding \t 5.06e+03 \t 0.76 \t 0.46\n",
      "cell_adhesion_mediator_activity \t 5.06e+03 \t 1.06 \t 0.65\n",
      "protein-macromolecule_adaptor_activity \t 3.24e+03 \t 4.4 \t 2.69\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ubiquitin_protein_ligase_binding \t 2.13e+03 \t 3.49 \t 2.13\n",
      "transferase_activity,_transferring_one-carbon_groups \t 5.06e+03 \t 0.91 \t 0.56\n",
      "inorganic_anion_transmembrane_transporter_activity \t 1.60e+03 \t 1.97 \t 1.21\n",
      "cytokine_activity \t 1.96e+03 \t 0.91 \t 0.56\n",
      "syntaxin_binding \t 3.91e+03 \t 1.06 \t 0.65\n",
      "cell_adhesion_molecule_binding \t 4.52e+03 \t 5.77 \t 3.53\n",
      "ubiquitin-like_protein_ligase_binding \t 2.64e+03 \t 3.64 \t 2.23\n",
      "histone_modifying_activity \t 2.11e+03 \t 0.76 \t 0.46\n",
      "cadherin_binding \t 3.07e+03 \t 3.64 \t 2.23\n",
      "phosphatidylinositol-3-phosphate_binding \t 8.27e+02 \t 0.76 \t 0.46\n",
      "endopeptidase_inhibitor_activity \t 3.32e+03 \t 0.61 \t 0.37\n",
      "general_transcription_initiation_factor_activity \t 5.06e+03 \t 0.0 \t 0.0\n",
      "3'-5'_exonuclease_activity \t 5.06e+03 \t 0.0 \t 0.0\n",
      "lipid_transfer_activity \t 1.04e+02 \t 0.0 \t 0.0\n",
      "calmodulin_binding \t 3.82e+02 \t 2.43 \t 1.48\n",
      "oxidoreductase_activity,_acting_on_NAD(P)H,_quinone_or_similar_compound_as_acceptor \t 5.06e+03 \t 0.0 \t 0.0\n",
      "polyubiquitin_modification-dependent_protein_binding \t 1.92e+03 \t 0.0 \t 0.0\n",
      "lipase_activity \t 2.52e+03 \t 1.67 \t 1.02\n",
      "DNA_helicase_activity \t 5.06e+03 \t 0.0 \t 0.0\n",
      "peptidase_activator_activity \t 5.06e+03 \t 0.91 \t 0.56\n",
      "electron_transfer_activity \t 5.06e+03 \t 0.3 \t 0.19\n",
      "protein_phosphorylated_amino_acid_binding \t 5.06e+03 \t 0.91 \t 0.56\n",
      "modified_amino_acid_binding \t 5.06e+03 \t 1.21 \t 0.74\n",
      "E-box_binding \t 4.49e+02 \t 0.91 \t 0.56\n",
      "myosin_phosphatase_activity \t 5.06e+03 \t 1.21 \t 0.74\n",
      "ATPase_binding \t 5.06e+03 \t 1.21 \t 0.74\n",
      "chloride_transmembrane_transporter_activity \t 1.98e+03 \t 1.52 \t 0.93\n",
      "phosphatase_binding \t 5.06e+03 \t 2.28 \t 1.39\n",
      "gamma-tubulin_binding \t 5.06e+03 \t 0.61 \t 0.37\n",
      "guanyl_nucleotide_binding \t 1.55e+03 \t 4.25 \t 2.6\n",
      "guanyl_ribonucleotide_binding \t 1.55e+03 \t 4.25 \t 2.6\n",
      "S-adenosylmethionine-dependent_methyltransferase_activity \t 5.06e+03 \t 0.61 \t 0.37\n",
      "methyltransferase_activity \t 5.06e+03 \t 0.91 \t 0.56\n",
      "phosphoprotein_binding \t 1.59e+03 \t 1.21 \t 0.74\n",
      "oxidoreductase_activity,_acting_on_CH-OH_group_of_donors \t 3.44e+03 \t 0.46 \t 0.28\n",
      "NADH_dehydrogenase_activity \t 5.06e+03 \t 0.0 \t 0.0\n",
      "NAD(P)H_dehydrogenase_(quinone)_activity \t 5.06e+03 \t 0.0 \t 0.0\n",
      "phosphatase_activity \t 3.03e+03 \t 3.03 \t 1.86\n",
      "endopeptidase_regulator_activity \t 5.06e+03 \t 0.76 \t 0.46\n",
      "serine-type_endopeptidase_inhibitor_activity \t 2.67e+03 \t 0.3 \t 0.19\n",
      "snRNA_binding \t 5.06e+03 \t 0.0 \t 0.0\n",
      "nuclear_receptor_binding \t 7.53e+02 \t 1.67 \t 1.02\n",
      "mRNA_3'-UTR_binding \t 2.67e+03 \t 0.3 \t 0.19\n",
      "glutamate_receptor_binding \t 1.17e+03 \t 0.76 \t 0.46\n",
      "molecular_function_activator_activity \t 2.55e+03 \t 11.23 \t 6.86\n",
      "monosaccharide_binding \t 7.04e+01 \t 0.15 \t 0.09\n",
      "calcium_channel_regulator_activity \t 1.77e+03 \t 0.76 \t 0.46\n",
      "exonuclease_activity \t 5.06e+03 \t 0.15 \t 0.09\n",
      "histone_binding \t 5.06e+03 \t 1.21 \t 0.74\n",
      "SMAD_binding \t 1.38e+03 \t 1.06 \t 0.65\n",
      "mitogen-activated_protein_kinase_binding \t 5.06e+03 \t 0.46 \t 0.28\n",
      "ATP_hydrolysis_activity \t 3.43e+03 \t 2.12 \t 1.3\n",
      "GTP_binding \t 1.83e+03 \t 3.95 \t 2.41\n",
      "chloride_channel_activity \t 2.51e+03 \t 1.06 \t 0.65\n",
      "signaling_receptor_complex_adaptor_activity \t 3.62e+03 \t 0.76 \t 0.46\n",
      "cell-cell_adhesion_mediator_activity \t 3.73e+03 \t 0.76 \t 0.46\n",
      "ubiquitin-like_protein_transferase_activity \t 1.45e+03 \t 2.73 \t 1.67\n",
      "calcium-dependent_protein_binding \t 2.51e+03 \t 1.06 \t 0.65\n",
      "structural_constituent_of_cytoskeleton \t 2.32e+03 \t 1.37 \t 0.83\n",
      "aminoacyltransferase_activity \t 1.92e+03 \t 2.88 \t 1.76\n",
      "nuclear_receptor_activity \t 5.06e+03 \t 0.76 \t 0.46\n",
      "ligand-activated_transcription_factor_activity \t 5.06e+03 \t 0.76 \t 0.46\n",
      "oxidoreductase_activity,_acting_on_the_CH-OH_group_of_donors,_NAD_or_NADP_as_acceptor \t 3.44e+03 \t 0.46 \t 0.28\n",
      "heat_shock_protein_binding \t 5.06e+03 \t 0.46 \t 0.28\n",
      "modification-dependent_protein_binding \t 3.73e+03 \t 0.76 \t 0.46\n",
      "SH3_domain_binding \t 2.41e+03 \t 1.52 \t 0.93\n",
      "ubiquitin-protein_transferase_activity \t 1.81e+03 \t 2.58 \t 1.58\n",
      "hormone_binding \t 3.80e+03 \t 1.06 \t 0.65\n",
      "molecular_carrier_activity \t 3.22e+03 \t 0.3 \t 0.19\n",
      "lyase_activity \t 8.52e+02 \t 0.91 \t 0.56\n",
      "protein_heterodimerization_activity \t 4.35e+03 \t 3.19 \t 1.95\n",
      "protein_tyrosine_phosphatase_activity \t 3.01e+03 \t 1.21 \t 0.74\n",
      "vitamin_binding \t 1.58e+03 \t 0.61 \t 0.37\n",
      "peptide_binding \t 5.06e+03 \t 3.34 \t 2.04\n",
      "3'-5'-RNA_exonuclease_activity \t 5.06e+03 \t 0.0 \t 0.0\n",
      "ADP_binding \t 1.92e+03 \t 0.0 \t 0.0\n",
      "ATP-dependent_chromatin_remodeler_activity \t 5.06e+03 \t 0.0 \t 0.0\n",
      "pre-mRNA_binding \t 1.92e+03 \t 0.0 \t 0.0\n",
      "ubiquitin-like_protein_conjugating_enzyme_binding \t 7.27e+02 \t 0.0 \t 0.0\n",
      "oxidoreduction-driven_active_transmembrane_transporter_activity \t 5.06e+03 \t 0.15 \t 0.09\n",
      "protein_transporter_activity \t 1.92e+03 \t 0.0 \t 0.0\n",
      "RNA_exonuclease_activity,_producing_5'-phosphomonoesters \t 5.06e+03 \t 0.0 \t 0.0\n",
      "nucleosomal_DNA_binding \t 5.06e+03 \t 0.0 \t 0.0\n",
      "protein_serine/threonine_kinase_activator_activity \t 5.06e+03 \t 0.15 \t 0.09\n",
      "nucleosome_binding \t 5.06e+03 \t 0.15 \t 0.09\n",
      "ATP-dependent_protein_folding_chaperone \t 5.06e+03 \t 0.0 \t 0.0\n",
      "RNA_polymerase_II_general_transcription_initiation_factor_activity \t 5.06e+03 \t 0.0 \t 0.0\n",
      "RNA_exonuclease_activity \t 5.06e+03 \t 0.0 \t 0.0\n",
      "RNA_polymerase_core_enzyme_binding \t 5.06e+03 \t 0.0 \t 0.0\n",
      "demethylase_activity \t 5.06e+03 \t 0.0 \t 0.0\n",
      "NADH_dehydrogenase_(ubiquinone)_activity \t 5.06e+03 \t 0.0 \t 0.0\n",
      "transition_metal_ion_transmembrane_transporter_activity \t 2.75e+02 \t 0.0 \t 0.0\n",
      "NADH_dehydrogenase_(quinone)_activity \t 5.06e+03 \t 0.0 \t 0.0\n",
      "steroid_binding \t 3.97e+03 \t 1.21 \t 0.74\n",
      "transmembrane_signaling_receptor_activity \t 7.74e+02 \t 9.1 \t 5.57\n",
      "thiolester_hydrolase_activity \t 3.32e+03 \t 0.61 \t 0.37\n",
      "translation_regulator_activity \t 5.06e+03 \t 0.76 \t 0.46\n",
      "hydrolase_activity,_acting_on_carbon-nitrogen_(but_not_peptide)_bonds,_in_linear_amides \t 3.63e+03 \t 0.91 \t 0.56\n",
      "translation_regulator_activity,_nucleic_acid_binding \t 5.06e+03 \t 0.46 \t 0.28\n",
      "protein_kinase_inhibitor_activity \t 5.06e+03 \t 0.91 \t 0.56\n",
      "DNA_binding \t 1.36e+02 \t 21.7 \t 13.27\n",
      "unfolded_protein_binding \t 5.06e+03 \t 0.46 \t 0.28\n",
      "acyltransferase_activity \t 3.37e+03 \t 4.7 \t 2.88\n",
      "acetylglucosaminyltransferase_activity \t 5.85e+02 \t 0.61 \t 0.37\n",
      "protein_serine/threonine/tyrosine_kinase_activity \t 5.06e+03 \t 0.61 \t 0.37\n",
      "virus_receptor_activity \t 5.06e+03 \t 0.91 \t 0.56\n",
      "amide_binding \t 3.81e+03 \t 3.95 \t 2.41\n",
      "exogenous_protein_binding \t 5.06e+03 \t 0.91 \t 0.56\n",
      "alpha-tubulin_binding \t 2.47e+03 \t 0.61 \t 0.37\n",
      "transcription_corepressor_binding \t 3.32e+03 \t 0.61 \t 0.37\n",
      "protein_tyrosine_kinase_binding \t 5.06e+03 \t 1.37 \t 0.83\n",
      "oxidoreductase_activity,_acting_on_NAD(P)H \t 3.22e+03 \t 0.3 \t 0.19\n",
      "endopeptidase_activity \t 3.59e+03 \t 2.73 \t 1.67\n",
      "translation_factor_activity,_RNA_binding \t 5.06e+03 \t 0.3 \t 0.19\n",
      "cytokine_receptor_binding \t 4.02e+03 \t 1.52 \t 0.93\n",
      "ubiquitin-like_protein_ligase_activity \t 1.61e+03 \t 2.12 \t 1.3\n",
      "monoatomic_anion_transmembrane_transporter_activity \t 2.97e+03 \t 1.52 \t 0.93\n",
      "proton_transmembrane_transporter_activity \t 3.07e+02 \t 1.52 \t 0.93\n",
      "hydrolase_activity,_acting_on_carbon-nitrogen_(but_not_peptide)_bonds \t 1.98e+03 \t 1.52 \t 0.93\n",
      "kinase_regulator_activity \t 4.30e+03 \t 2.73 \t 1.67\n",
      "ABC-type_transporter_activity \t 5.06e+03 \t 0.61 \t 0.37\n",
      "ligase_activity,_forming_carbon-nitrogen_bonds \t 1.54e+03 \t 0.61 \t 0.37\n",
      "intramolecular_oxidoreductase_activity \t 1.54e+03 \t 0.61 \t 0.37\n",
      "alcohol_binding \t 5.06e+03 \t 1.06 \t 0.65\n",
      "sulfur_compound_binding \t 3.59e+03 \t 2.73 \t 1.67\n",
      "monoatomic_anion_channel_activity \t 2.51e+03 \t 1.06 \t 0.65\n",
      "transcription_coregulator_activity \t 1.69e+03 \t 3.19 \t 1.95\n",
      "cytokine_receptor_activity \t 3.91e+03 \t 1.06 \t 0.65\n",
      "protein_phosphatase_regulator_activity \t 2.51e+03 \t 1.06 \t 0.65\n",
      "pyrophosphatase_activity \t 3.64e+03 \t 6.98 \t 4.27\n",
      "vitamin_B6_binding \t 5.06e+03 \t 0.15 \t 0.09\n",
      "oxidoreductase_activity,_acting_on_peroxide_as_acceptor \t 2.84e+03 \t 0.15 \t 0.09\n",
      "RNA_polymerase_binding \t 5.06e+03 \t 0.15 \t 0.09\n",
      "DNA_nuclease_activity \t 5.06e+03 \t 0.15 \t 0.09\n",
      "basal_RNA_polymerase_II_transcription_machinery_binding \t 5.06e+03 \t 0.15 \t 0.09\n",
      "basal_transcription_machinery_binding \t 5.06e+03 \t 0.15 \t 0.09\n",
      "growth_factor_activity \t 5.06e+03 \t 1.67 \t 1.02\n",
      "peptidase_inhibitor_activity \t 1.33e+03 \t 0.91 \t 0.56\n",
      "copper_ion_binding \t 3.62e+03 \t 0.76 \t 0.46\n",
      "heparin_binding \t 1.96e+03 \t 1.82 \t 1.11\n",
      "protein_homodimerization_activity \t 2.62e+03 \t 6.53 \t 3.99\n",
      "nucleocytoplasmic_carrier_activity \t 5.06e+03 \t 0.0 \t 0.0\n",
      "miRNA_binding \t 5.06e+03 \t 0.0 \t 0.0\n",
      "snoRNA_binding \t 5.06e+03 \t 0.0 \t 0.0\n",
      "macromolecule_transmembrane_transporter_activity \t 1.92e+03 \t 0.0 \t 0.0\n",
      "hydrolase_activity,_acting_on_acid_anhydrides,_in_phosphorus-containing_anhydrides \t 3.64e+03 \t 6.98 \t 4.27\n",
      "hydrolase_activity,_acting_on_acid_anhydrides \t 3.64e+03 \t 6.98 \t 4.27\n",
      "organophosphate_ester_transmembrane_transporter_activity \t 1.92e+03 \t 0.0 \t 0.0\n",
      "tRNA_methyltransferase_activity \t 5.06e+03 \t 0.0 \t 0.0\n",
      "serine-type_peptidase_activity \t 5.06e+03 \t 1.06 \t 0.65\n",
      "RNA_polymerase_II_complex_binding \t 5.06e+03 \t 0.0 \t 0.0\n",
      "ubiquitin-protein_transferase_regulator_activity \t 5.06e+03 \t 0.46 \t 0.28\n",
      "zinc_ion_binding \t 4.12e+03 \t 7.44 \t 4.55\n",
      "oxidoreductase_activity,_acting_on_the_CH-CH_group_of_donors,_NAD_or_NADP_as_acceptor \t 5.06e+03 \t 0.46 \t 0.28\n",
      "14-3-3_protein_binding \t 1.48e+03 \t 0.46 \t 0.28\n",
      "tRNA_binding \t 2.67e+03 \t 0.3 \t 0.19\n",
      "RNA_helicase_activity \t 3.22e+03 \t 0.3 \t 0.19\n",
      "RNA_endonuclease_activity \t 5.06e+03 \t 0.3 \t 0.19\n",
      "ATP-dependent_activity,_acting_on_RNA \t 3.22e+03 \t 0.3 \t 0.19\n",
      "molecular_function_inhibitor_activity \t 1.45e+03 \t 5.01 \t 3.06\n",
      "ribonucleoside_triphosphate_phosphatase_activity \t 2.71e+03 \t 6.37 \t 3.9\n",
      "phosphotransferase_activity,_phosphate_group_as_acceptor \t 5.06e+03 \t 0.46 \t 0.28\n",
      "palmitoyltransferase_activity \t 5.06e+03 \t 0.46 \t 0.28\n",
      "SNAP_receptor_activity \t 3.44e+03 \t 0.46 \t 0.28\n",
      "receptor_tyrosine_kinase_binding \t 3.83e+03 \t 0.91 \t 0.56\n",
      "kinase_inhibitor_activity \t 5.06e+03 \t 0.91 \t 0.56\n",
      "kinase_activator_activity \t 2.41e+03 \t 1.52 \t 0.93\n",
      "ubiquitin_protein_ligase_activity \t 1.61e+03 \t 2.12 \t 1.3\n",
      "protein_kinase_regulator_activity \t 2.78e+03 \t 2.28 \t 1.39\n",
      "frizzled_binding \t 3.44e+03 \t 0.46 \t 0.28\n",
      "lipid_kinase_activity \t 5.06e+03 \t 0.46 \t 0.28\n",
      "small_molecule_sensor_activity \t 5.06e+03 \t 0.46 \t 0.28\n",
      "sodium_ion_transmembrane_transporter_activity \t 4.07e+03 \t 1.52 \t 0.93\n",
      "primary_active_transmembrane_transporter_activity \t 1.96e+03 \t 0.91 \t 0.56\n",
      "beta-tubulin_binding \t 1.48e+03 \t 0.46 \t 0.28\n",
      "ligase_activity,_forming_carbon-sulfur_bonds \t 5.06e+03 \t 0.46 \t 0.28\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "transferase_activity \t 5.06e+03 \t 19.42 \t 11.87\n",
      "ubiquitin_binding \t 3.44e+03 \t 0.46 \t 0.28\n",
      "N-methyltransferase_activity \t 1.48e+03 \t 0.46 \t 0.28\n",
      "nucleobase-containing_compound_kinase_activity \t 5.06e+03 \t 0.46 \t 0.28\n",
      "active_transmembrane_transporter_activity \t 4.32e+03 \t 2.88 \t 1.76\n",
      "G_protein_activity \t 5.06e+03 \t 0.46 \t 0.28\n",
      "cysteine-type_endopeptidase_regulator_activity_involved_in_apoptotic_process \t 5.06e+03 \t 0.46 \t 0.28\n",
      "intramembrane_lipid_transporter_activity \t 5.06e+03 \t 0.46 \t 0.28\n",
      "signal_sequence_binding \t 2.84e+03 \t 0.15 \t 0.09\n",
      "oxidoreductase_activity,_acting_on_the_aldehyde_or_oxo_group_of_donors \t 7.89e+02 \t 0.15 \t 0.09\n",
      "cis-trans_isomerase_activity \t 5.06e+03 \t 0.15 \t 0.09\n",
      "cyclin-dependent_protein_serine/threonine_kinase_regulator_activity \t 7.89e+02 \t 0.15 \t 0.09\n",
      "carbohydrate_derivative_transmembrane_transporter_activity \t 5.06e+03 \t 0.15 \t 0.09\n",
      "metalloendopeptidase_activity \t 3.91e+03 \t 1.06 \t 0.65\n",
      "nucleobase-containing_compound_transmembrane_transporter_activity \t 5.06e+03 \t 0.15 \t 0.09\n",
      "translation_initiation_factor_activity \t 5.06e+03 \t 0.15 \t 0.09\n",
      "pentosyltransferase_activity \t 5.06e+03 \t 0.15 \t 0.09\n",
      "general_transcription_initiation_factor_binding \t 5.06e+03 \t 0.15 \t 0.09\n",
      "quaternary_ammonium_group_binding \t 3.63e+02 \t 0.15 \t 0.09\n",
      "serine_hydrolase_activity \t 5.06e+03 \t 1.21 \t 0.74\n",
      "peptidase_activity \t 4.43e+03 \t 4.4 \t 2.69\n",
      "hydrolase_activity \t 3.59e+03 \t 20.33 \t 12.43\n",
      "pyridoxal_phosphate_binding \t 5.06e+03 \t 0.15 \t 0.09\n",
      "histone_deacetylase_binding \t 1.58e+03 \t 0.61 \t 0.37\n",
      "transition_metal_ion_binding \t 5.06e+03 \t 9.56 \t 5.84\n",
      "nuclear_receptor_coactivator_activity \t 1.54e+03 \t 0.61 \t 0.37\n",
      "RNA_polymerase_II-specific_DNA-binding_transcription_factor_binding \t 2.93e+03 \t 3.19 \t 1.95\n",
      "ubiquitin-like_protein_peptidase_activity \t 2.11e+03 \t 0.76 \t 0.46\n",
      "glycosaminoglycan_binding \t 3.47e+03 \t 2.28 \t 1.39\n",
      "oxidoreductase_activity,_acting_on_the_CH-CH_group_of_donors \t 5.06e+03 \t 0.61 \t 0.37\n",
      "immune_receptor_activity \t 5.06e+03 \t 1.37 \t 0.83\n",
      "promoter-specific_chromatin_binding \t 3.32e+03 \t 0.61 \t 0.37\n",
      "heterocyclic_compound_binding \t 1.02e+03 \t 46.13 \t 28.2\n",
      "p53_binding \t 5.06e+03 \t 0.3 \t 0.19\n",
      "metal_cluster_binding \t 2.67e+03 \t 0.3 \t 0.19\n",
      "iron-sulfur_cluster_binding \t 2.67e+03 \t 0.3 \t 0.19\n",
      "transferase_activity,_transferring_sulphur-containing_groups \t 1.89e+03 \t 0.3 \t 0.19\n",
      "carbohydrate_binding \t 3.32e+03 \t 1.82 \t 1.11\n",
      "lipid_transporter_activity \t 1.44e+03 \t 1.06 \t 0.65\n",
      "serine-type_endopeptidase_activity \t 3.80e+03 \t 1.06 \t 0.65\n",
      "monocarboxylic_acid_binding \t 5.06e+03 \t 0.76 \t 0.46\n",
      "hydrolase_activity,_acting_on_ester_bonds \t 4.06e+03 \t 6.53 \t 3.99\n",
      "metallopeptidase_activity \t 4.11e+03 \t 1.67 \t 1.02\n",
      "cysteine-type_peptidase_activity \t 5.06e+03 \t 1.21 \t 0.74\n",
      "dioxygenase_activity \t 3.44e+03 \t 0.46 \t 0.28\n",
      "organic_acid_binding \t 1.68e+03 \t 1.37 \t 0.83\n",
      "N-acyltransferase_activity \t 4.49e+02 \t 0.91 \t 0.56\n",
      "organic_cyclic_compound_binding \t 9.31e+02 \t 46.89 \t 28.66\n",
      "exopeptidase_activity \t 3.83e+03 \t 0.91 \t 0.56\n",
      "phosphatase_regulator_activity \t 3.80e+03 \t 1.06 \t 0.65\n",
      "solute:monoatomic_cation_symporter_activity \t 1.38e+03 \t 1.06 \t 0.65\n",
      "translation_repressor_activity \t 2.67e+03 \t 0.3 \t 0.19\n",
      "secondary_active_transmembrane_transporter_activity \t 5.06e+03 \t 1.82 \t 1.11\n",
      "5'-3'_RNA_polymerase_activity \t 5.06e+03 \t 0.3 \t 0.19\n",
      "RNA_polymerase_activity \t 5.06e+03 \t 0.3 \t 0.19\n",
      "G_protein-coupled_peptide_receptor_activity \t 2.62e+03 \t 0.76 \t 0.46\n",
      "peptide_receptor_activity \t 2.62e+03 \t 0.76 \t 0.46\n",
      "protein-disulfide_reductase_activity \t 5.06e+03 \t 0.3 \t 0.19\n",
      "protein_kinase_activator_activity \t 1.09e+03 \t 1.21 \t 0.74\n",
      "ubiquitin_conjugating_enzyme_activity \t 5.06e+03 \t 0.3 \t 0.19\n",
      "signaling_receptor_regulator_activity \t 2.75e+03 \t 4.1 \t 2.5\n",
      "GTPase_binding \t 1.58e+02 \t 2.73 \t 1.67\n",
      "heme_binding \t 1.70e+03 \t 1.21 \t 0.74\n",
      "nucleotidyltransferase_activity \t 5.07e+02 \t 1.37 \t 0.83\n",
      "carboxylic_ester_hydrolase_activity \t 2.32e+03 \t 1.37 \t 0.83\n",
      "integrin_binding \t 5.06e+03 \t 1.37 \t 0.83\n",
      "FAD_binding \t 7.89e+02 \t 0.15 \t 0.09\n",
      "DNA_endonuclease_activity \t 5.06e+03 \t 0.15 \t 0.09\n",
      "aminopeptidase_activity \t 2.84e+03 \t 0.15 \t 0.09\n",
      "DNA_polymerase_activity \t 5.06e+03 \t 0.15 \t 0.09\n",
      "ligase_activity,_forming_carbon-oxygen_bonds \t 5.06e+03 \t 0.15 \t 0.09\n",
      "aminoacyl-tRNA_ligase_activity \t 5.06e+03 \t 0.15 \t 0.09\n",
      "4_iron,_4_sulfur_cluster_binding \t 5.06e+03 \t 0.15 \t 0.09\n",
      "regulatory_RNA_binding \t 5.06e+03 \t 0.15 \t 0.09\n",
      "peptidyl-prolyl_cis-trans_isomerase_activity \t 5.06e+03 \t 0.15 \t 0.09\n",
      "neutral_L-amino_acid_transmembrane_transporter_activity \t 5.06e+03 \t 0.15 \t 0.09\n",
      "ligase_activity \t 3.07e+02 \t 1.52 \t 0.93\n",
      "protein_phosphatase_inhibitor_activity \t 5.06e+03 \t 0.46 \t 0.28\n",
      "oxidoreductase_activity,_acting_on_paired_donors,_with_incorporation_or_reduction_of_molecular_oxygen \t 3.01e+03 \t 1.21 \t 0.74\n",
      "tumor_necrosis_factor_receptor_superfamily_binding \t 1.48e+03 \t 0.46 \t 0.28\n",
      "hexosyltransferase_activity \t 4.78e+02 \t 1.67 \t 1.02\n",
      "phosphatase_inhibitor_activity \t 5.06e+03 \t 0.46 \t 0.28\n",
      "protein_kinase_A_binding \t 4.61e+02 \t 0.46 \t 0.28\n",
      "carboxylic_acid_binding \t 1.68e+03 \t 1.37 \t 0.83\n",
      "NAD_binding \t 5.06e+03 \t 0.46 \t 0.28\n",
      "NADP_binding \t 3.44e+03 \t 0.46 \t 0.28\n",
      "signaling_receptor_activator_activity \t 1.80e+03 \t 3.79 \t 2.32\n",
      "O-acyltransferase_activity \t 5.06e+03 \t 0.46 \t 0.28\n",
      "monocarboxylic_acid_transmembrane_transporter_activity \t 5.06e+03 \t 0.3 \t 0.19\n",
      "transferase_activity,_transferring_alkyl_or_aryl_(other_than_methyl)_groups \t 5.06e+03 \t 0.3 \t 0.19\n",
      "histone_methyltransferase_activity \t 5.06e+03 \t 0.3 \t 0.19\n",
      "L-amino_acid_transmembrane_transporter_activity \t 5.06e+03 \t 0.3 \t 0.19\n",
      "ATP-dependent_activity \t 4.43e+03 \t 4.25 \t 2.6\n",
      "glycosyltransferase_activity \t 1.28e+03 \t 2.28 \t 1.39\n",
      "GDP_binding \t 5.06e+03 \t 0.61 \t 0.37\n",
      "double-stranded_RNA_binding \t 5.06e+03 \t 0.61 \t 0.37\n",
      "G_protein-coupled_receptor_binding \t 9.53e+01 \t 2.12 \t 1.3\n",
      "amino_acid_transmembrane_transporter_activity \t 5.06e+03 \t 0.46 \t 0.28\n",
      "single-stranded_RNA_binding \t 5.06e+03 \t 0.76 \t 0.46\n",
      "hydrolase_activity,_hydrolyzing_O-glycosyl_compounds \t 5.06e+03 \t 0.61 \t 0.37\n",
      "acetyltransferase_activity \t 1.54e+03 \t 0.61 \t 0.37\n",
      "monooxygenase_activity \t 5.06e+03 \t 0.91 \t 0.56\n",
      "ubiquitin-like_protein_binding \t 5.06e+03 \t 0.76 \t 0.46\n",
      "chromatin_DNA_binding \t 2.11e+03 \t 0.76 \t 0.46\n",
      "antiporter_activity \t 5.06e+03 \t 0.76 \t 0.46\n",
      "deubiquitinase_activity \t 2.11e+03 \t 0.76 \t 0.46\n",
      "cysteine-type_endopeptidase_activity \t 3.73e+03 \t 0.76 \t 0.46\n",
      "single-stranded_DNA_binding \t 3.62e+03 \t 0.76 \t 0.46\n",
      "receptor_ligand_activity \t 1.80e+03 \t 3.79 \t 2.32\n",
      "cytokine_binding \t 3.01e+03 \t 1.21 \t 0.74\n",
      "protease_binding \t 1.70e+03 \t 1.21 \t 0.74\n",
      "UDP-glycosyltransferase_activity \t 4.78e+02 \t 1.21 \t 0.74\n",
      "symporter_activity \t 8.39e+02 \t 1.21 \t 0.74\n",
      "tetrapyrrole_binding \t 1.70e+03 \t 1.21 \t 0.74\n",
      "iron_ion_binding \t 3.80e+03 \t 1.06 \t 0.65\n",
      "carboxylic_acid_transmembrane_transporter_activity \t 2.51e+03 \t 1.06 \t 0.65\n",
      "molecular_transducer_activity \t 1.02e+03 \t 11.99 \t 7.33\n",
      "signaling_receptor_activity \t 1.02e+03 \t 11.99 \t 7.33\n",
      "extracellular_matrix_structural_constituent \t 5.06e+03 \t 1.37 \t 0.83\n",
      "catalytic_activity \t 4.18e+03 \t 45.22 \t 27.64\n",
      "transcription_corepressor_activity \t 3.93e+03 \t 1.37 \t 0.83\n",
      "magnesium_ion_binding \t 5.06e+03 \t 1.82 \t 1.11\n",
      "acyltransferase_activity,_transferring_groups_other_than_amino-acyl_groups \t 4.10e+03 \t 1.82 \t 1.11\n",
      "organic_anion_transmembrane_transporter_activity \t 1.38e+03 \t 1.67 \t 1.02\n",
      "active_monoatomic_ion_transmembrane_transporter_activity \t 3.24e+03 \t 1.97 \t 1.21\n",
      "small_GTPase_binding \t 7.25e+01 \t 2.28 \t 1.39\n",
      "hydro-lyase_activity \t 5.06e+03 \t 0.46 \t 0.28\n",
      "metalloexopeptidase_activity \t 2.21e+03 \t 0.46 \t 0.28\n",
      "manganese_ion_binding \t 3.44e+03 \t 0.46 \t 0.28\n",
      "sulfur_compound_transmembrane_transporter_activity \t 2.67e+03 \t 0.3 \t 0.19\n",
      "peptide_N-acetyltransferase_activity \t 2.67e+03 \t 0.3 \t 0.19\n",
      "flavin_adenine_dinucleotide_binding \t 2.47e+03 \t 0.61 \t 0.37\n",
      "phospholipid_transporter_activity \t 3.44e+03 \t 0.46 \t 0.28\n",
      "ATPase-coupled_transmembrane_transporter_activity \t 1.77e+03 \t 0.76 \t 0.46\n",
      "neurotransmitter_receptor_activity \t 4.56e+02 \t 0.76 \t 0.46\n",
      "amino_acid_binding \t 5.06e+03 \t 0.46 \t 0.28\n",
      "antioxidant_activity \t 5.06e+03 \t 0.61 \t 0.37\n",
      "oxidoreductase_activity,_acting_on_a_sulfur_group_of_donors \t 5.06e+03 \t 0.3 \t 0.19\n",
      "methylation-dependent_protein_binding \t 3.32e+03 \t 0.61 \t 0.37\n",
      "sterol_binding \t 3.44e+03 \t 0.46 \t 0.28\n",
      "lysine_N-methyltransferase_activity \t 1.48e+03 \t 0.46 \t 0.28\n",
      "nuclear_thyroid_hormone_receptor_binding \t 5.06e+03 \t 0.15 \t 0.09\n",
      "protein-lysine_N-methyltransferase_activity \t 1.48e+03 \t 0.46 \t 0.28\n",
      "ubiquitin-like_protein_conjugating_enzyme_activity \t 3.22e+03 \t 0.3 \t 0.19\n",
      "hydrolase_activity,_hydrolyzing_N-glycosyl_compounds \t 3.22e+03 \t 0.3 \t 0.19\n",
      "N-acetyltransferase_activity \t 1.48e+03 \t 0.46 \t 0.28\n",
      "Hsp70_protein_binding \t 5.06e+03 \t 0.3 \t 0.19\n",
      "cysteine-type_deubiquitinase_activity \t 2.11e+03 \t 0.76 \t 0.46\n",
      "methylated_histone_binding \t 5.06e+03 \t 0.46 \t 0.28\n",
      "carbon-oxygen_lyase_activity \t 5.06e+03 \t 0.46 \t 0.28\n",
      "cyclin-dependent_protein_serine/threonine_kinase_activity \t 2.84e+03 \t 0.15 \t 0.09\n",
      "DNA-directed_5'-3'_RNA_polymerase_activity \t 5.06e+03 \t 0.15 \t 0.09\n",
      "cyclin-dependent_protein_kinase_activity \t 2.84e+03 \t 0.15 \t 0.09\n",
      "protein_serine/threonine_kinase_inhibitor_activity \t 3.63e+02 \t 0.15 \t 0.09\n",
      "endonuclease_activity,_active_with_either_ribo-_or_deoxyribonucleic_acids_and_producing_5'-phosphomonoesters \t 2.67e+03 \t 0.3 \t 0.19\n",
      "core_promoter_sequence-specific_DNA_binding \t 5.06e+03 \t 0.3 \t 0.19\n",
      "enzyme-substrate_adaptor_activity \t 5.06e+03 \t 0.3 \t 0.19\n",
      "ATPase_regulator_activity \t 5.06e+03 \t 0.3 \t 0.19\n",
      "protein_methyltransferase_activity \t 5.06e+03 \t 0.61 \t 0.37\n",
      "protein-folding_chaperone_binding \t 5.06e+03 \t 1.06 \t 0.65\n",
      "peptide-lysine-N-acetyltransferase_activity \t 2.67e+03 \t 0.3 \t 0.19\n",
      "ubiquitin_ligase-substrate_adaptor_activity \t 5.06e+03 \t 0.3 \t 0.19\n",
      "protein_tyrosine/serine/threonine_phosphatase_activity \t 3.22e+03 \t 0.3 \t 0.19\n",
      "cholesterol_binding \t 1.89e+03 \t 0.3 \t 0.19\n",
      "ATPase-coupled_monoatomic_cation_transmembrane_transporter_activity \t 1.05e+03 \t 0.3 \t 0.19\n",
      "sulfotransferase_activity \t 1.89e+03 \t 0.3 \t 0.19\n",
      "telomeric_DNA_binding \t 5.06e+03 \t 0.3 \t 0.19\n",
      "RNA_endonuclease_activity,_producing_5'-phosphomonoesters \t 2.67e+03 \t 0.3 \t 0.19\n",
      "disulfide_oxidoreductase_activity \t 5.06e+03 \t 0.3 \t 0.19\n",
      "translation_initiation_factor_binding \t 5.06e+03 \t 0.15 \t 0.09\n",
      "protein_carrier_chaperone \t 2.84e+03 \t 0.15 \t 0.09\n",
      "organic_acid_transmembrane_transporter_activity \t 1.70e+03 \t 1.21 \t 0.74\n",
      "phosphatidylinositol_phosphate_phosphatase_activity \t 5.06e+03 \t 0.15 \t 0.09\n",
      "histone_acetyltransferase_activity \t 2.67e+03 \t 0.3 \t 0.19\n",
      "Hsp90_protein_binding \t 2.67e+03 \t 0.3 \t 0.19\n",
      "carbon-carbon_lyase_activity \t 3.22e+03 \t 0.3 \t 0.19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "hydrolase_activity,_acting_on_glycosyl_bonds \t 3.91e+03 \t 1.06 \t 0.65\n",
      "histone_H3_methyltransferase_activity \t 2.67e+03 \t 0.3 \t 0.19\n",
      "isomerase_activity \t 8.39e+02 \t 1.21 \t 0.74\n",
      "DNA_secondary_structure_binding \t 5.06e+03 \t 0.15 \t 0.09\n",
      "deacetylase_activity \t 5.06e+03 \t 0.15 \t 0.09\n",
      "peptidase_regulator_activity \t 3.12e+03 \t 1.82 \t 1.11\n",
      "transcription_coactivator_activity \t 6.53e+02 \t 2.12 \t 1.3\n",
      "enzyme_inhibitor_activity \t 1.69e+03 \t 3.19 \t 1.95\n",
      "chromatin_binding \t 2.72e+03 \t 4.7 \t 2.88\n"
     ]
    }
   ],
   "source": [
    "for term in goDict1:\n",
    "    observed = [goDict2[term], count2-goDict2[term]]\n",
    "    expected = [goDict1[term], count1-goDict1[term]]\n",
    "    p = fisher_exact([observed,expected])[1]\n",
    "    termNew = term.replace(\" \",\"_\")\n",
    "    padj = p * len(goDict1)\n",
    "    if goDict5[term] >= 25:\n",
    "        print(termNew, \"\\t\", '{:.2e}'.format(padj), \"\\t\", round(goDict1[term]/count2*100,2), \"\\t\", round(goDict1[term]/count1*100,2))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3feece27",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
